### ISSUE TITLE
Error: You have invalid characters in your hostname.

### ISSUE THREAD

**Issue by jbcreek** [poweradmin/poweradmin#378]
While trying to add a domain:

Error: You have invalid characters in your hostname.
Error: test.net failed - Invalid hostname.

CentOS 8
php-fpm-7.4.7-1.el8.remi.x86_64
php-pecl-mcrypt-1.0.3-1.el8.remi.7.4.x86_64
php-sodium-7.4.7-1.el8.remi.x86_64
php-json-7.4.7-1.el8.remi.x86_64
php-xml-7.4.7-1.el8.remi.x86_64
php-mbstring-7.4.7-1.el8.remi.x86_64
php-intl-7.4.7-1.el8.remi.x86_64
php-pdo-7.4.7-1.el8.remi.x86_64
php-cli-7.4.7-1.el8.remi.x86_64
php-7.4.7-1.el8.remi.x86_64
php-gd-7.4.7-1.el8.remi.x86_64
php-opcache-7.4.7-1.el8.remi.x86_64
php-mysqlnd-7.4.7-1.el8.remi.x86_64
php-common-7.4.7-1.el8.remi.x86_64
oniguruma5php-6.9.5+rev1-2.el8.remi.x86_64
php-pecl-zip-1.19.0-1.el8.remi.7.4.x86_64
mysql-server-8.0.17-3.module_el8.0.0+181+899d6349.x86_64
pdns-4.3.0-2.el8.x86_64
pdns-backend-mysql-4.3.0-2.el8.x86_64

Selinux is disabled.

Thanks in advance,

Jeff

**Comment by bryanpedini** [comment]
Same issue here:

Error: You have invalid characters in your hostname.
Error: anything.tld failed - Invalid hostname.

CentOS 8 (Core)
nginx 1.14.1 + certbot-auto SSL reverse proxy on httpd 2.4.37 as per [this guide](https://www.tecmint.com/install-powerdns-poweradmin-mariadb-in-centos-rhel/2/). (added nginx and remi-repo with PHP 7.4)
pdns 4.3.0
pdns-backend-mysql 4.3.0

`/etc/pdns/pdns.conf` content:
```
launch=gmysql
gmysql-host=127.0.0.1
gmysql-user=powerdns
gmysql-password=<db-password>
gmysql-dbname=powerdns
local-address=<ip-address>
local-port=53
setgid=pdns
setuid=pdns
```

**Comment by bryanpedini** [comment]
Debian 10 buster new virtual machine:

first, pdns-server is outdated, maintained package is still on 4.1, if needed
then, even after following [this guide](https://varunver.wordpress.com/2019/10/11/install-php-mcrypt-on-debian-10-php-7-3/), the admin panel still reports "Error: You have to install PHP mcrypt extension!".

I basically cannot even try to reproduce the "invalid characters" issue, maybe I'll consider opening a new issue when I have more time.

**Comment by nodenineinc** [comment]
Follow the notes in the prior semi-related case to get mcrypt functioning for php 7.3:
https://github.com/poweradmin/poweradmin/issues/376

Enabling mcrypt for php <7.3 is well documented around the 'net.

The 'invalid characters' error is likely related to the bug I filed a few weeks ago if you are on php7.3.  Same condition is happening.
https://github.com/poweradmin/poweradmin/issues/381

**Comment by bryanpedini** [comment]
@nodenineinc Doesn't work with PHP 7.4 either, and 7.2 is in "critical security patches only" support mode, I won't install it unless it is absolutely critical for something I absolutely need to have...
I ended up installing pdns-resolver on CentOS 8 with PowerDNS-Admin in Python Flask in a Docker container and abandoned poweradmin, too much hassle for no good when there was a ready-to-work literally plug-and-play Docker image for the Flask app (and the GUI is beautiful too, instead of poweradmin).

You can continue solving this issue, but even tho I have created it, I moved on to another product since I needed an admin interface 'round that time I filed the bug, and this was not working for me at all!

**Comment by edmondas** [comment]
It can be related to the deprecated mcrypt library. That was fixed with PR #337.
